app/src/sharedTest/java/org/oppia/android/app/devoptions/platformparameters/PlatformParametersFragmentTest.kt

```diff
@@ -418,6 +423,229 @@ class PlatformParametersFragmentTest {
+        expectedDisplayName = SPLASH_SCREEN_WELCOME_MSG_PARAMETER_NAME
+      )
+    }
+  }
+
+  @Test
+  fun testPlatformParametersFragment_boolParam_onlyOverriddenValue_returnsOverriddenSyncStatus() {
+    executeInPreviousAppInstance { testApplicationComponent ->
+      addTestBooleanOverriddenPlatformParameterToDatabase(testApplicationComponent, true)
+      testApplicationComponent.getTestCoroutineDispatchers().runCurrent()
+    }
+    setUpTestApplicationComponent()
+    launch(PlatformParametersTestActivity::class.java).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(0)
+      verifyPlatformParameterSyncStatus(
+        position = 0,
+        expectedSyncStatus = context.getString(
+          R.string.platform_parameter_overridden_sync_status
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testPlatfromParametersFragment_boolParam_onlyOverriddenValue_returnsOverriddenBackground() {
+    executeInPreviousAppInstance { testApplicationComponent ->
+      addTestBooleanOverriddenPlatformParameterToDatabase(testApplicationComponent, true)
+      testApplicationComponent.getTestCoroutineDispatchers().runCurrent()
+    }
+    setUpTestApplicationComponent()
+    launch(PlatformParametersTestActivity::class.java).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(0)
+      verifyPlatformParameterBackgroundColor(
+        position = 0,
+        expectedColor = 0xFFBE563C.toInt()
+      )
+    }
+  }
+
+  @Test
+  fun testPlatformParametersFragment_boolParam_withRemoteAndOverridden_returnsOverriddenStatus() {
+    executeInPreviousAppInstance { component ->
+      addTestRemoteBooleanPlatformParameterToDatabase(component, true)
+      addTestBooleanOverriddenPlatformParameterToDatabase(component, false)
+      component.getTestCoroutineDispatchers().runCurrent()
+    }
+    setUpTestApplicationComponent()
+    launch(PlatformParametersTestActivity::class.java).use {
+      testCoroutineDispatchers.runCurrent()
+
+      scrollToPosition(0)
+      verifyPlatformParameterSyncStatus(
+        position = 0,
+        expectedSyncStatus = context.getString(
+          R.string.platform_parameter_overridden_sync_status
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testPlatformParametersFragment_boolParam_withRemoteAndOverridden_returnsOverriddenValue() {
+    executeInPreviousAppInstance { component ->
+      addTestRemoteBooleanPlatformParameterToDatabase(component, true)
+      addTestBooleanOverriddenPlatformParameterToDatabase(component, false)
+      component.getTestCoroutineDispatchers().runCurrent()
+    }
+    setUpTestApplicationComponent()
+    launch(PlatformParametersTestActivity::class.java).use {
+      testCoroutineDispatchers.runCurrent()
+
+      scrollToPosition(0)
+      val splashScreenWelcomeMsgParameter = getEphemeralPlatformParameters()[0]
+      verifyPlatformParameterState(
+        position = 0,
+        expectedValue = splashScreenWelcomeMsgParameter.currentValue
```

Comment:
This currently feels like we’re testing `actual == actual`, since the UI is already driven by the ephemeral state. For remote / override cases, let’s instead assert against the true expected value to ensure correctness.

```suggestion
        expectedValue = PlatformParameterValue.newBuilder()
            .setBoolean(false)
            .build()
```

Reply:
updated